src/queueservice/MongoDBProxy.py

```diff
@@ -1,12 +1,116 @@
+import re
+import uuid
+import json
+from datetime import datetime
+from pymongo import MongoClient
 
+# Helper class for mongodb
+class MongoAPI:
 
+    # Creates client to connect to Mongo DB
+    def __init__(self, url: str, password: str):
+        
+        # Load url with password
+        cluster_key = re.sub("<password>", password, url)
 
-class MongoDBProxy:
-    def __init__(self, url):
-        self.url = url
+        # Create client and connect to database 'queues'
+        self.client = MongoClient(cluster_key)
+        self.db = self.client.get_database('test_queue_service') # !!! Hardcoded Element !!!
 
-    def query(self, query_string):
-        pass
+    # Retrieve the queue_array with the specified uuid
+    def get_queue_for_category(self, id: uuid) -> list:
 
-    def update(self, query_string, update_item):
-        pass
\ No newline at end of file
+        # Connect to queue table in the database
+        queue = self.db.get_collection('queue') # !!! Hardcoded Element !!!
+        
+        # Retrive document (row) with specified id
+        document = queue.find_one({"id": id})
+
+        # Check if document with id exists in table
+        if document:
+            return document["queue"]
+
+        return "Queue not found with specified id."
+    
+    # Replace the queue in the specified uuid with the queue_array
+    def update_queue_for_category(self, id: uuid, queue_array: list):
+
+        # Connect to queue table in the database
+        queue = self.db.get_collection('queue') # !!! Hardcoded Element !!!
+
+        # # Define new value to be pushed to queue array
+        # new_value = {"$push": new_queue_entry}
+
+        # Update the queue with specified id
+        queue.update_one({"id": id}, queue_array)
```

Comment:
Have you had an opportunity to test this? I'm unfamiliar with pymongo but does this update the array in the document or replace the document with the array?